pkg/controller/orchestrator/orchestrator_controller_test.go

```diff
@@ -137,27 +150,41 @@ var _ = Describe("Orchestrator controller", func() {
 			mgr, err := manager.New(cfg, manager.Options{})
 			Expect(err).NotTo(HaveOccurred())
 			c = mgr.GetClient()
-			recFn, requests = SetupTestReconcile(newReconciler(mgr))
+			recFn, requests = SetupTestReconcile(newReconciler(mgr, orcClient))
 			Expect(add(mgr, recFn)).To(Succeed())
 			stop = StartTestManager(mgr)
 
 			// wait a second for a request
 			Consistently(requests, reconcileTime).ShouldNot(Receive(Equal(expectedRequest)))
-			Eventually(requests).Should(Receive(Equal(expectedRequest)))
+			Eventually(requests, afterReconcileTime).Should(Receive(Equal(expectedRequest)))
 		})
 
 		It("should unregister cluster when deleting it from kubernetes", func() {
 			// delete the cluster
 			Expect(c.Delete(context.TODO(), cluster)).To(Succeed())
 
-			// wait a second for a request
-			// This is the request thats removes the finalizers from cluster
-			Consistently(requests, reconcileTime).ShouldNot(Receive(Equal(expectedRequest)))
-			Eventually(requests).Should(Receive(Equal(expectedRequest)))
+			// wait two seconds for a request
+			// This is the request thats unregister cluster from orchestrator
+			Eventually(requests, "1.2s").Should(Receive(Equal(expectedRequest)))
```

Comment:
Let's make this a multiple of `reconcileTime`. Also is there any reason for `1.2s`?